benchmarks/dynamo/pr_time_benchmarks/cirron/cirron.py

```diff
@@ -0,0 +1,108 @@
+import os
+from ctypes import byref, c_int, c_uint64, CDLL, POINTER, Structure
+from subprocess import call
+
+from pkg_resources import resource_filename
+
+
+class Counter(Structure):
+    _fields_ = [
+        ("time_enabled_ns", c_uint64),
+        ("instruction_count", c_uint64),
+        ("branch_misses", c_uint64),
+        ("page_faults", c_uint64),
+    ]
+
+    def __str__(self):
+        return self.__repr__()
+
+    def __repr__(self):
+        repr = "Counter("
+
+        for field, _ in Counter._fields_:
+            repr += f"{field}={getattr(self, field)}, "
+        repr = repr[:-2] + ")"
+
+        return repr
+
+
+lib_path = resource_filename(__name__, "cirronlib.so")
+if not os.path.exists(lib_path):
+    source_path = resource_filename(__name__, "cirronlib.cpp")
+    print("cirronlib is not compiled yet, compiling it now...")
+    print(
+        f"calling c++ -std=c++17 -O3 -shared -fPIC -o {lib_path} {source_path}",
+    )
+    exit_status = call(
+        f"c++ -std=c++17 -O3 -shared -fPIC -o {lib_path} {source_path}", shell=True
+    )
+    if exit_status != 0:
+        raise Exception(
+            "Failed to compile cirronlib.cpp, make sure you have 'c++' installed."
+        )
```

Comment:
wth is this lol. Use cpp_extension or better yet put it in the main torch/csrc

Reply:
This is how cirron.py is implemented originally I ported as is, I talked to 
@aorenste and he gave me pointers on how torch handles c++ functions i guess I will create new functions
under torch._C and call them. probably we will doing a "clean room implementation" at that point